go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -295,9 +249,29 @@ func (bsnc *BaseSecondaryNetworkController) addLogicalPortToNetworkForNAD(pod *k
 			pod.Namespace, pod.Name, nadName, time.Since(start), libovsdbExecuteTime)
 	}()
 
-	ops, lsp, podAnnotation, newlyCreated, err := bsnc.addLogicalPortToNetwork(pod, nadName, network)
-	if err != nil {
-		return err
+	// local pods need a logical port configured for them
+	// but remote pods on layer2 topologies with interconnect also need a remote
+	// logical port
+	isLocalPod := bsnc.isPodScheduledinLocalZone(pod)
+	requiresLogicalPort := isLocalPod || bsnc.isLayer2Interconnect()
+
+	var err error
+	var podAnnotation *util.PodAnnotation
+	var ops []ovsdb.Operation
+	var lsp *nbdb.LogicalSwitchPort
+	var newlyCreated bool
+	if requiresLogicalPort {
```

Comment:
maybe I'm missing something, but, what is the scenario for a pod attachment that does **not** require the logical port created?

Reply:
Only local pods or remote pods in layer2 topology require a port.

So to answer your question, any remote pod for any toplogy that is not layer2. So remote pods for default, layer3 or localnet networks.

Reply:
Thank you.

I still am confused - afaiu, those topologies also require logical ports. What's so special about **this** particular logical port type ? What does it have the others don't ? (I haven't checked the code in depth, maybe I should).

Reply:
They do require logical ports for pods local to the zone, but not for pods remote to the zone. this is an interconnect specific issue.

For default and layer3 networks, inter-connectivity between the nodes is achieved connecting each zone cluster routers through the transit switch, so on every zone you have to create a remote port for every remote zone cluster router in the transit switch. In regards to pods, you just create ports for local pods in node specific switch.

For localnet networks, there is no remote connectivity required through the OVN netwokr as it is provided by the physical network.

For layer2 networks, which are flat have a single switch and have no router, that single switch becomes the transit switch and you basically have to create ports there on every zone for every pod. For remote pods, the port is of type remote instead of a default port type.

Reply:
yeah so it's the comment that is confusing me:
> 	// local pods need a logical port configured for them
	// but remote pods on layer2 topologies with interconnect also need a remote
	// logical port

basically, **all** pods require the logical port. Independently of the topology. Thus, I don't understand why it is behind an if clause.

Am I making sense ?

Reply:
Remote pods **do not require** a logical port except for layer2 networks. 

I don't know how to better explain this. We can have a chat if you want.

Reply:
reworded the comment

and added similar comments in the other places where we are doing a similar consideration